lib/util.js

```diff
@@ -430,6 +431,38 @@ module.exports.timeout = (ms) => {
 module.exports.serverSwaggerPluginId = 'org.zowe.zlux'
 module.exports.agentSwaggerPluginId = 'org.zowe.zlux.agent';
 
+module.exports.getCookieName = (port) => {
+  const zoweInstanceId = process.env['ZOWE_INSTANCE'];
+  const id = isHaMode() ? zoweInstanceId : String(port);
+  return 'connect.sid.' + id;
+}
+
+const isHaMode = module.exports.isHaMode = function () {
+  const zoweInstanceId = process.env['ZOWE_INSTANCE'];
+  const zoweHaInstanceId = process.env['ZWELS_HA_INSTANCE_ID'];
+  return zoweInstanceId && zoweHaInstanceId;
+}
+
+module.exports.generateHaSessionId = function (request) {
+  const TOKEN_NAME = 'apimlAuthenticationToken';
```

Comment:
this shouldnt be here, its too specific to apiml.
this function should be in sso-auth.
the server should query auth plugins for the capability "generateHaSessionId" and if any exist, call the first one.
If none exist, return `return crypto.randomBytes(24).toString('hex');`